experimental/packages/opentelemetry-exporter-prometheus/src/PrometheusExporter.ts

```diff
@@ -87,8 +87,11 @@ export class PrometheusExporter extends MetricReader {
 
     if (config.preventServerStart !== true) {
       this.startServer()
-        .then(callback)
-        .catch(err => diag.error(err));
+        .then(() => callback?.())
+        .catch(err => {
+          diag.error(err);
+          callback?.(err);
+        });
```

Comment:
callback may be called twice here if startServer succeeds but the callback fails

```suggestion
        .then(
         () => callback?.(),
         (err) => {
           diag.error(err);
           callback?.(err);
         });
```

Reply:
Good catch, thanks for pointing this out. should be fixed in [c87043f](https://github.com/open-telemetry/opentelemetry-js/pull/3552/commits/c87043fda1b8a05453d32ba0941c0e8e14d481ae) :slightly_smiling_face: